test/helpers/config_scanning.rb

```diff
@@ -13,6 +13,7 @@ module ConfigScanning
       REGISTER_CALLBACK_PATTERN = /register_callback\(:['"]?([a-z\._]+)['"]?\)/
       NAMED_DEPENDENCY_PATTERN = /^\s*named[ (]+\:?([a-z0-9\._]+).*$/
       EVENT_BUFFER_MACRO_PATTERN = /(capacity_key|enabled_key)\s+:['"]?([a-z\._]+)['"]?/
+      ASSIGNED_CONSTANT_PATTERN = /[A-Z]+\s*=\s*:['"]?([a-z\._]+)['"]?\s*/
```

Comment:
A minor issue (predating this PR) for these patterns is that we're using `['"]` twice instead of a backreference for the first match. This technically permits `key = "value'` with unmatched quotes. I think with it only being used in the tests, we're ok for now.

Reply:
Thanks for pointing this out. I updated the patterns to use backreferences in 1aa3e5709649f1998a9696f306bdcf47f16f9708